charts/osschallenge/templates/pvc.yaml

```
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ include "osschallenge.fullname" . }}
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.persistence.storageClassName }}
+  storageClassName: {{ .Values.persistence.storageClassName }}
+  {{- end }}
+  accessModes:
+  - ReadWriteOnce
```

the `replicaCount` would be obsolete at this point.

---

So you mean, I should set replicaCount to 0 then? And why would it be obsolete?

---

Not to 0, but having `ReadWriteOnce` hard coded here, would not make it possible to have more than one replica running on different nodes. And if replica aren't running on different nodes, there's really no need for it if the use case would be HA.

---

> The access modes are:
    ReadWriteOnce -- the volume can be mounted as read-write by a single node
    ReadOnlyMany -- the volume can be mounted read-only by many nodes
    ReadWriteMany -- the volume can be mounted as read-write by many nodes

---

Of course the storageProvider needs to support `RWX`. 

https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

---

something like this:
```suggestion
  {{ .Values.persistance.accessModes | toYaml | nindent 2 }}
```

with values:
```yaml
persistance:
  accessModes:
  - ReadWriteOnce
```

---

persistence !

---

oops